homeassistant/components/teletask/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""Connects to Teletask platform."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PORT, EVENT_HOMEASSISTANT_STOP
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ["pyteletask==0.1.2"]
+
+DOMAIN = "teletask"
+
+ATTR_DISCOVER_DEVICES = "devices"
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TELETASK_CONFIG = ""
+DATA_TELETASK = "data_teletask"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {vol.Required(CONF_HOST): cv.string, vol.Required(CONF_PORT): cv.port}
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
```

Comment:
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Reply:
@frenck Hey Frenck,

I've been looking at these 2 Design Decisions while doing some refactoring.
At this point I've managed to create the Platform on-boarding via the config_flow however I don't have the ability to dynamicly discover the devices themselves (so I'd need to either have an endpoint that allows me to manually add Devices into the devices list, or I need something that can read config from file).

Do you have any example of any of the integrations that do it like this?

Br
T